.github/workflows/full-build.yml

```diff
@@ -50,44 +50,45 @@ jobs:
     name: Linux (Wheel) with Python ${{ matrix.python_version }}
     needs: lint
     runs-on: ubuntu-22.04
+    container: fedora:39
     timeout-minutes: 30
     if: "!contains(github.event.head_commit.message, 'skip ci')"
     strategy:
       matrix:
-        python_version: ['3.10', '3.11', '3.12']
+        python_version: ['3.11', '3.12']
     outputs:
       targz: gaphor-${{ steps.install.outputs.version }}.tar.gz
       wheel: gaphor-${{ steps.install.outputs.version }}-py3-none-any.whl
       version: ${{ steps.install.outputs.version }}
     steps:
-      - name: Harden Runner
-        uses: step-security/harden-runner@1b05615854632b887b69ae1be8cbefe72d3ae423 # v2.6.0
-        with:
-          egress-policy: audit
-
-      - uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4.1.0
-        with:
-          ref: ${{ github.event.pull_request.head.sha }}
       - name: Install Linux Dependencies
         run: >
-          sudo apt-get update -qq && sudo apt-get install -qq --no-install-recommends upx
-          gir1.2-gtk-4.0 libgirepository1.0-dev libgtksourceview-5-dev libadwaita-1-dev
-          graphviz
-      - name: Set up Python
-        uses: actions/setup-python@65d7f2d534ac1bc67fcd62888c5f4f3d2cb2b236 # v4.7.1
+          dnf install -y gcc git gobject-introspection-devel
+          cairo-gobject-devel graphviz pkg-config gtk4 gtksourceview5-devel
+          libadwaita-devel cairo-devel upx python${{ matrix.python_version }}
+          python${{ matrix.python_version }}-devel python-launcher
+          xorg-x11-server-Xvfb
+      - name: Install pip
+        run: python${{ matrix.python_version }} -m ensurepip
+      - name: Install pipx
+        run: |
+          py -${{ matrix.python_version }} -m pip install pipx
+          echo "/github/home/.local/bin" >> $GITHUB_PATH
+      - uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4.1.0
         with:
-          python-version: ${{ matrix.python_version }}
-          allow-prereleases: true
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Set ownership of Checkout directory
+        run: chown -R $(id -u):$(id -g) $PWD
```

Comment:
Same here. I'd probably add it to the `Install Linux Dependencies` step.

Reply:
In the Linux dependencies step we need to install Git for the checkout dependency to work properly, and then finally set the correct permissions. So we need these 3 steps separately to keep this order.